doc/design/externally-managed-pf-reconcile-optimization.md

```diff
@@ -0,0 +1,374 @@
+* Maintain backward compatibility with existing deployments
+
+### Non-Goals
+
+* Changing how externally managed PFs validate their prerequisites (numVfs, MTU, link type)
+* Adding new user-facing configuration options for this behavior
+
+## Proposal
+
+Three options are proposed to address this issue. All options aim to remove the unconditional
+`return true` for externally managed PFs while ensuring VF Admin MAC configuration is correct.
+
+### Option 1: Extend API / Report Admin MAC in Node State
+
+Extend the SriovNetworkNodeState API to report the VF Admin MAC in the status, and use this
+information to make intelligent reconciliation decisions.
+
+#### Workflow Description
+
+1. During device discovery, the config daemon reads and reports each VF's Admin MAC address
+   in the SriovNetworkNodeState status.
+2. When checking if reconfiguration is needed (`NeedToUpdateSriov()`), trigger reconfiguration for any ethernet interface (including externally managed) if the VF Admin MAC is empty or uninitialized.
+3. Remove the unconditional `return true` for externally managed interfaces.
+4. Update `CheckStatusChanges()` in the generic plugin to include externally managed interfaces
+   in the comparison logic.
+
+#### API Extensions
+
+Add a new field to the `VirtualFunction` struct in `api/v1/sriovnetworknodestate_types.go`:
+
+```go
+type VirtualFunction struct {
+    Name            string `json:"name,omitempty"`
+    Mac             string `json:"mac,omitempty"`
+    AdminMac        string `json:"adminMac,omitempty"`  // NEW: Administrative MAC address
+    Assigned        string `json:"assigned,omitempty"`
+    Driver          string `json:"driver,omitempty"`
+    PciAddress      string `json:"pciAddress"`
+    Vendor          string `json:"vendor,omitempty"`
+    DeviceID        string `json:"deviceID,omitempty"`
+    Vlan            int    `json:"Vlan,omitempty"`
+    Mtu             int    `json:"mtu,omitempty"`
+    VfID            int    `json:"vfID"`
+    VdpaType        string `json:"vdpaType,omitempty"`
+    RepresentorName string `json:"representorName,omitempty"`
+    GUID            string `json:"guid,omitempty"`
+}
+```
+
+#### Implementation Details
+
+1. **Discovery Changes** (`pkg/host/internal/sriov/sriov.go`):
+   - In `DiscoverSriovDevices()`, read the VF Admin MAC from the PF using netlink
+   - Populate the new `AdminMac` field in the VirtualFunction status
+
+2. **Reconciliation Logic Changes** (`api/v1/helper.go`):
+   - Remove the unconditional `return true` for externally managed PFs
+   - Add Admin MAC check for Ethernet netdevice VFs.
+   ```go
+   // Check if Admin MAC needs to be configured
+   if strings.EqualFold(ifaceStatus.LinkType, consts.LinkTypeETH) {
+       if groupSpec.DeviceType == "" || groupSpec.DeviceType == consts.DeviceTypeNetDevice {
+           if vfStatus.AdminMac == "" || vfStatus.AdminMac == consts.UninitializedMac {
```

Comment:
vfStatus.AdminMac comes from the DiscoverSriov procedure. What value is expected for `UninitializedMac` ? 00:00:00:00:00:00 ?

Reply:
> 00:00:00:00:00:00

yes